lib/checkbufferoverrun.cpp

```diff
@@ -554,20 +554,27 @@ ValueFlow::Value CheckBufferOverrun::getBufferSize(const Token *bufTok) const
             return *value;
     }
 
-    if (!var)
+    MathLib::bigint dim = -1;
+    if (var) {
+        dim = std::accumulate(var->dimensions().cbegin(), var->dimensions().cend(), 1LL, [](MathLib::bigint i1, const Dimension &dim) {
+            return i1 * dim.num;
+        });
+    }
+    else if (bufTok->tokType() == Token::Type::eString) {
+        dim = bufTok->str().size() - 1; // plus terminating 0 minus two "
```

Comment:
It's better to use `Token::getStrLength` that should handle backslashes properly.